app/controllers/course/reference_timelines_controller.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+class Course::ReferenceTimelinesController < Course::ComponentController
+  load_and_authorize_resource :reference_timeline, through: :course
+
+  def create
+    if @reference_timeline.save
+      render partial: 'reference_timeline', locals: {
+        reference_timeline: @reference_timeline,
+        render_times: reference_timeline_params[:reference_times_attributes].present?
+      }
+    else
+      render json: { errors: @reference_timeline.errors }, status: :bad_request
+    end
+  end
+
+  def update
+    if @reference_timeline.update(reference_timeline_update_params)
+      head :ok
+    else
+      render json: { errors: @reference_timeline.errors }, status: :bad_request
+    end
+  end
+
+  def destroy
+    @alternative_timeline_id = destroy_params[:revert_to]
+    revert_course_users_to_alternative_timeline if @alternative_timeline_id.present?
+
+    ActiveRecord::Base.transaction do
+      if @updated_course_users.present?
+        CourseUser.import! @updated_course_users, on_duplicate_key_update: [:reference_timeline_id]
+        @reference_timeline.course_users.reload
+      end
+
+      @reference_timeline.destroy!
+      head :ok
+    end
+  rescue ActiveRecord::InvalidForeignKey # @alternative_timeline_id is invalid
+    head :bad_request
+  rescue StandardError
+    render json: { errors: @reference_timeline.errors }, status: :bad_request
+  end
+
+  private
+
+  def reference_timeline_params
+    params.require(:reference_timeline).permit(:title, :weight, {
+      reference_times_attributes: Course::ReferenceTimesController.base_params
```

Comment:
I originally added this param here because I envisage Timelines Designer to cache new timelines and the corresponding times before POSTing it to the back-end. That makes sense, because the UI I was thinking needs a _Save changes_ button, i.e., no auto-save.

But I noted your advice. Should I decide to implement auto-save, I will remove this param.